<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="UTF-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<title>Task 04</title>
</head>

<body>
	<pre>
Задание:
Создайте пользовательский класс исключения DivideByZeroError
Сделайте так, чтобы функция divide при попытке деления на 0 выбрасывала эту ошибку.
Добавьте код, обрабатывающий исключение.
---------------------------
class DivideByZeroError extends Error {
  constructor() {
    super('Ошибка: попытка деления на 0!');
    this.name = 'DivideByZeroError'
  }
}

function divide(operand1, operand2) {
  // if (operand2 == 0) throw Error(`!деление на 0!`);
  if (operand2 == 0) throw new DivideByZeroError;
  return operand1 / operand2;
}

console.log(divide(10, 5));
console.log(divide(100, 0));
console.log(divide(1000, 50));
_______________________________________________________________________
</pre>
	<script>
		/*
		Задание:
		Создайте пользовательский класс исключения DivideByZeroError
		Сделайте так, чтобы функция divide при попытке деления на 0 выбрасывала эту ошибку.
		Добавьте код, обрабатывающий исключение.
		*/

		class DivideByZeroError extends Error {
			constructor(nameOperand) {
				super(`Ошибка: попытка деления на 0! (имя операнда ${nameOperand})`);
				this.name = 'DivideByZeroError'
			}

		}

		function divide(operand1, operand2) {
			// if (operand2 == 0) throw Error(`!деление на 0!`);
			if (operand2 == 0) throw new DivideByZeroError('operand2');
			return operand1 / operand2;
		}

		console.log(divide(10, 5));
		console.log(divide(100, 0));
		console.log(divide(1000, 50));

	</script>
</body>

</html>